docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Ranking/Ranking.ttinclude

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+<# if (TrainerOptions != null) { #>
+<#=OptionsInclude#>
+<# } #>
+
+namespace Samples.Dynamic.Trainers.Ranking
+{
+    public static class <#=ClassName#>
+    {<#=Comments#>
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            var dataPoints = GenerateRandomDataPoints(1000);
```

Comment:
> var dataPoints = GenerateRandomDataPoints(1000); [](start = 11, length = 49)

There is an asymmetry between the dataset loading for train and test.

I would prefer to have one call to `GenerateRandomDataPoints`, followed by a call to `LoadFromEnumerable`, and then followed by a `TrainTestSplit` that uses the `GroupingColumn` (or whatever we called it) option. This would be a good place to remind people that they need to be careful about the GroupId column.

Reply:
You are right, I like your suggested approach better, I am going to write an issue about it, and fix it for all the samples (not just ranking).
This template is pretty standard across the different samples at the moment.

---
In reply to: [275909222](https://github.com/dotnet/machinelearning/pull/3338#discussion_r275909222) [](ancestors = 275909222)